services/venice-controller/src/main/java/com/linkedin/venice/controller/util/ParentControllerConfigUpdateUtils.java

```diff
@@ -0,0 +1,171 @@
+package com.linkedin.venice.controller.util;
+
+import com.linkedin.venice.controller.VeniceControllerClusterConfig;
+import com.linkedin.venice.controller.VeniceParentHelixAdmin;
+import com.linkedin.venice.controller.kafka.protocol.admin.UpdateStore;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.meta.Store;
+import com.linkedin.venice.schema.SchemaEntry;
+import com.linkedin.venice.schema.writecompute.WriteComputeSchemaConverter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import org.apache.avro.Schema;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * This class is a utility class for Parent Controller store update logics.
+ * The method here aims to take in current status and request params to determine if certain feature is updated / should
+ * be updated based on some customized logics.
+ */
+public class ParentControllerConfigUpdateUtils {
+  public static final Logger LOGGER = LogManager.getLogger(ParentControllerConfigUpdateUtils.class);
+  public static final WriteComputeSchemaConverter updateSchemaConverter = WriteComputeSchemaConverter.getInstance();
+
+  /**
+   * This method takes in current status and request and try to determine whether to change partial update config.
+   * The check logic is:
+   * Step (1): If there is explict request, we will respect the request and maybe update config if new request value is
+   * different from existing config value. In this step, if we are enabling partial update, we will also perform a dry
+   * run to validate schema. If validation fails, it will throw exception and fail the whole request.
+   * Step (2): If there is NO explict request and store is being converted into hybrid store, we will check the cluster
+   * config and store's latest A/A config to see whether we should by default enable partial update. If so, we will also
+   * perform a dry on to validate schema. If validation fails, it will swallow the exception and log warning message. It
+   * will not turn on partial update and will not fail the whole request.
+   */
+  public static boolean checkAndMaybeApplyPartialUpdateConfig(
+      VeniceParentHelixAdmin parentHelixAdmin,
+      String clusterName,
+      String storeName,
+      Optional<Boolean> partialUpdateRequest,
+      UpdateStore setStore,
+      boolean storeBeingConvertedToHybrid) {
+    Store currentStore = parentHelixAdmin.getVeniceHelixAdmin().getStore(clusterName, storeName);
+    VeniceControllerClusterConfig clusterConfig =
+        parentHelixAdmin.getVeniceHelixAdmin().getHelixVeniceClusterResources(clusterName).getConfig();
+    boolean partialUpdateConfigChanged = false;
+    setStore.writeComputationEnabled = currentStore.isWriteComputationEnabled();
+    if (partialUpdateRequest.isPresent()) {
+      setStore.writeComputationEnabled = partialUpdateRequest.get();
```

Comment:
So even the update param is same as the current setup, we want to pass it to child controller, right?
This is fine and just wondering the reason, and is it considering that child controller may be using a different config value for the same field in the same store?

Reply:
Honestly, I'm not sure what's the exact original reason to do this. This is actually the original config update logic that exists before my change in this PR, and we even have a check at the end to make sure every update store must have something to change, otherwise it is considered invalid and will throw.

I think, as you pointed out, it does give a chance to uniformly override a config value that might be inconsistent between parent and child region and between different child regions. We can consider evolving this logic in the future if we think it is not necessary, but to change this there will be some extra changes, I think it could be done in future PR though.

Reply:
SG.